app/controllers/sidebar_sections_controller.rb

```diff
@@ -56,6 +56,21 @@ def update
     render json: failed_json, status: 403
   end
 
+  def reorder
+    sidebar_section = SidebarSection.find_by(id: reorder_params["sidebar_section_id"])
+    @guardian.ensure_can_edit!(sidebar_section)
+
+    order = reorder_params["links_order"].map(&:to_i).each_with_index.to_h
+    links = sidebar_section.sidebar_section_links.sort_by { |link| order[link.linkable_id] }
+    sidebar_section.sidebar_section_links.upsert_all(
+      links.map.with_index { |link, index| link.attributes.merge(position: index) },
```

Comment:
I believe `links` is already an array here so I'm not sure if we need `map`. Also, I think we can merge the array operations here directly into line 64.

Reply:
We need map to return enumerator which has `with_index` method 

https://www.bootrails.com/blog/ruby-map-with-index/

But yeah, I will move it to line 64 as it is a better place